R/html_dependencies.R

```diff
@@ -55,9 +65,60 @@ html_dependency_bootstrap <- function(theme) {
       # These shims are necessary for IE 8 compatibility
       "shim/html5shiv.min.js",
       "shim/respond.min.js"),
-    stylesheet = paste0("css/", theme, ".min.css"))
+    stylesheet = paste0("css/", theme, ".min.css"),
+    # CSS rules yanked from inst/rmd/h/default.html that should remain for historical
+    # reasons, but shouldn't be included if bslib is relevant
+    head = format(tags$style(HTML(
+      "h1 {font-size: 34px;}
+       h1.title {font-size: 38px;}
+       h2 {font-size: 30px;}
+       h3 {font-size: 24px;}
+       h4 {font-size: 18px;}
+       h5 {font-size: 16px;}
+       h6 {font-size: 12px;}
+       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
+       pre:not([class]) { background-color: white }"
+    )))
+  )
 }
 
+bootstrap_dependencies <- function(theme) {
+  deps <- html_dependency_bootstrap(theme)
+  if (inherits(deps, "html_dependency")) list(deps) else deps
+}
+
+# At the moment, theme may be either NULL (no Bootstrap), a string (Bootswatch
+# 3 name), a bslib::bs_theme(), or a list of arguments to bs_theme().
+as_bs_theme <- function(theme) {
+  if (is_bs_theme(theme)) {
+    return(theme)
+  }
+  if (is.list(theme)) {
+    return(do.call(bslib::bs_theme, theme))
+  }
+  NULL
+}
+
+is_bs_themeish <- function(theme) {
+  is_bs_theme(theme) || is.list(theme)
+}
+
+is_bs_theme <- function(theme) {
+  is_available("bslib") &&
+    bslib::is_bs_theme(theme)
+}
+
+theme_version <- function(theme) {
+  if (is_bs_theme(theme)) {
+    bslib::theme_version(theme)
+  } else if (is.list(theme)) {
+    theme$version %||% bslib::version_default()
+  } else {
+    substr(html_dependency_bootstrap("default")$version, 1, 1)
+  }
+}
```

Comment:
This is a place that will error if **bslib** is not installed. I saw that when trying a very simple example with the default template Rmd. I added this in the header and with no **bslib** installed : 
````yaml
output: 
  html_document:
    theme:
      bootswatch: flatly
````
and got this 
```r
Error in loadNamespace(name) : there is no package called 'bslib'
Calls: <Anonymous> ... loadNamespace -> withRestarts -> withOneRestart -> doWithOneRestart
Exécution arrêtée
```
because `bslib::version_default()` is called without checks on the package existence. 

I think we should check and fail early if **bslib** features are asked for in `theme` but the package is not installed.